python/loyalty_workflow.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.common import WorkflowIDReusePolicy
+from temporalio.exceptions import WorkflowAlreadyStartedError
+from temporalio.workflow import ParentClosePolicy
+from temporalio.client import WorkflowHandle
+
+import email_strings
+from shared import *
```

Comment:
```suggestion
with workflow.unsafe.imports_passed_through():
    from shared import Customer
```

And add other classes explicitly. We want to not reload the shared module every workflow run, and also we should explicitly state which classes/types we import